OSDOCS-6081: adds known issue to RNs for 4.13 (#60041)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

* There is a disk discovery delay when attaching storage to workloads. (link:https://issues.redhat.com/browse/OCPBUGS-11149[*OCPBUGS-11149*])

> * When upgrading from {product-title} 4.12 to 4.13, Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. The renaming causes `SriovNetworkNodePolicy` object to not be able to locate the original interface. Consequently, virtual functions (VFs) cannot be created due to the interface not being found.

Reviewer: ```suggestion
* When upgrading from {product-title} 4.12 to 4.13, Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. Renaming the network node policies causes the `SriovNetworkNodePolicy` object to not be able to locate the original interface. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.
```

> * When upgrading from {product-title} 4.12 to 4.13, the Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. Renaming the network node policies prevents the `SriovNetworkNodePolicy` object from locating the original interface. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.

Reviewer 2: I would expand on: "Renaming the network node policies prevents the `SriovNetworkNodePolicy` object from locating the original interface" => only when the 'pfNames' field is used in the policy to specify the interface name such as `ens7f0`.

Reviewer 2: @zhaozhanqi This is your understanding as well, right?

Reviewer 3: yes, I added one comment as below.

> * When upgrading from {product-title} 4.12 to 4.13, Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. Renaming the network node policies causes the `SriovNetworkNodePolicy` object to not be able to locate the original interface. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.

Reviewer 4: Should NIC be written out since this content is not near the networking content?
```suggestion
* When upgrading from {product-title} 4.12 to 4.13, the Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. Renaming the network node policies prevents the `SriovNetworkNodePolicy` object from locating the original interface. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.
```

> * When upgrading from {product-title} 4.12 to 4.13, the Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0`. Renaming the network node policies prevents the `SriovNetworkNodePolicy` object from locating the original interface only when the `pfNames` field is used in the policy such as `ens7f0`. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.

Reviewer 3: the Mellanox NIC will be renamed such as `ens7f0` to `ens7f0np0` after upgrade to rhel9 kernel. since SriovNetworkNodePolicy was using the original nic name(ens7f0) before upgrade. So after upgrade the VF may cannot be created because the interface cannot be found

The workarounds is edit the `SriovNetworkNodePolicy` CR and add another interface name. eg

Given the original `SriovNetworkNodePolicy` was created before upgrade. like below
```
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: cx6dx
  namespace: openshift-sriov-network-operator
spec:
  deviceType: netdevice
  nicSelector:
    deviceID: "101d"
    pfNames:
      - ens7f0
    vendor: '15b3'
  nodeSelector:
    feature.node.kubernetes.io/sriov-capable: 'true'
  numVfs: 4
  mtu: 9000
  priority: 99
  resourceName: cx6dx
```

After upgrade if find the interface name `ens7f0` renamed to `ens7f0np0`. here we can edit the SriovNetworkNodePolicy CR by `oc edit SriovNetworkNodePolicy cx6dx` and then add another interface name `ens7f0np0`. 

```
  deviceType: netdevice
  nicSelector:
    deviceID: "101d"
    pfNames:
      - ens7f0
      - ens7f0np0
    vendor: '15b3'
  nodeSelector:
    feature.node.kubernetes.io/sriov-capable: 'true'
  numVfs: 4

```

Reviewer 3: cc @wizhaoredhat @bn222 @vrindle

Reviewer 5: Need to state that we recommend adding  `- ens7f0np0` before the upgrade.

Reviewer 5: /remove-lgtm

Reviewer 3: @bn222  adding `- ens7f0np0` before upgrade or after upgrade, both two are worked in my testing.

+

> To work around this issue, you must manually delete and re-add a SR-IOV network node policy with the new name after updating to {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-11677[*OCPBUGS-13186*])

Reviewer 4: ```suggestion
To work around this issue, you must manually delete and re-add an SR-IOV network node policy with the new name after updating to {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-11677[*OCPBUGS-13186*])
```

Reviewer 4: Oh - just realized after I submitted. The link text doesn't match the URL (`OCPBUGS-11677[*OCPBUGS-13186*]`)

> To work around this issue, users should manually delete and re-add a SR-IOV network node policy with the new name when updating to {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-11677[*OCPBUGS-13186*])

Reviewer: ```suggestion
To work around this issue, you must manually delete and re-add a SR-IOV network node policy with the new w name when updating to {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-11677[*OCPBUGS-13186*])
```

Do you need to do this after you update or before you update? Or do we not know?

* There is a disk discovery delay when attaching storage to workloads. (link:https://issues.redhat.com/browse/OCPBUGS-11149[*OCPBUGS-11149*])

> * When upgrading from {product-title} 4.12 to 4.13, the Mellanox NIC renames SR-IOV network node policies such as `ens7f0` to `ens7f0np0` after upgrading to RHEL 9 kernel. Renaming the network node policies prevents the `SriovNetworkNodePolicy` object from locating the original interface only when the `pfNames` field is used in the policy such as `ens7f0`. Consequently, virtual functions (VFs) cannot be created because the interface cannot be found.

Reviewer 5: When upgrading from {product-title} 4.12 to 4.13, the Mellanox NIC gets renamed as a consequence of moving to RHEL 9 kernel. The SR-IOV network node policies should take this renaming into account. For example, if `ens7f0` is referenced from the policy, add `ens7f0np0` too before upgrading.

Reviewer 2: Should we be more specific and say RHEL 9.2 kernel?

+
To work around this issue, you must manually edit the `SriovNetworkNodePolicy` custom resource (CR) to add `ens7f0np0` before updating to {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-13186[*OCPBUGS-13186*])

> The following provides an example of the policy updates with both names being added to `SriovNetworkNodePolicy` to ensure compability:

Reviewer: ```suggestion
The following code provides an example of the policy updates with both names being added to `SriovNetworkNodePolicy` to ensure compatibility:
```

Reviewer 6: 😅 I need another cup of coffee this morning it seems

+

> [source,terminal]

Reviewer: This should probably be `[source,yaml]`.

----

>   deviceType: netdevice

Reviewer: The newest YAML guidelines say that sections of a file need to be enclosed in `# ...` (for YAML). This looks like the middle of a YAML excerpt, so it probably needs one at the start and another at the end.
